sdk/playwrighttesting/microsoft-playwright-testing/review/microsoft-playwright-testing.api.md

```diff
@@ -0,0 +1,95 @@
+## API Report File for "@azure/microsoft-playwright-testing"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import type { TokenCredential } from '@azure/identity';
```

Comment:
```suggestion
import type { TokenCredential } from "@azure/core-auth"
```

(of course you'd want to update the actual import and then regenerate the API Extractor review file, not just apply the suggestion)

Reply:
Any specific reason we want this change? We already have taken a dependency on `@azure/identity` to provide out of the box support for `DefaultAzureCredentials` to improve customer's onboarding experience.

Adding another dependency just for types (that is already provided by `@azure/identity`) will just increase the bundle size.

Reply:
Great question! 

Our client libraries are meant to be agnostic of the TokenCredential implementation. Customers could theoretically provide their own implementation or use a version of `@azure/identity` of their choosing. They may also decide to use DefaultAzureCredential in dev-mode and ManagedIdentityCredential in production so we should not force them one way or the other (I don't think we're doing that here from what I see which is good)

We do this by having our client libraries accept a `TokenCredential` type from `@azure/core-auth` which keeps our dependencies on core, allowing customers to bring in their own dependency on identity. While it's true that `@azure/identity` _also_ re-exports the TokenCredential interface that is for consumers of your client library to use when they want to create a custom credential. 

I took a look at your usages of `@azure/identity`. Outside of examples (which, by the way, should use DefaultAzureCredential - _not_ AzureCliCredential as they currently do) nothing here makes use of `@azure/identity`'s concrete credentials.

Here's what you should do:

1. Move `@azure/identity` from `dependencies` to `devDependencies`
2. Replaces all `import type { TokenCredential } from "@azure/identity"` with `import type { TokenCredential } from "@azure/core-auth"`
3. Add `@azure/core-auth` to your `dependencies` if not already added
4. Replace AzureCliCredential with DefaultAzureCredential in all your samples 

The net-result should be the same for you, but a more flexible interface for your customers who may be using older versions of `@azure/identity` (or newer major versions in the future). Let me know if that's not the case!